src/managers/prompt.ts

```diff
@@ -0,0 +1,886 @@
+import { BasicTool, BasicOptions } from "../basic";
+import { ManagerTool } from "../basic";
+const base = new BasicTool()
+var Zotero = base.getGlobal("Zotero")
+/**
+ * Prompt for setting up or executing some commands quickly.
+ * 
+ * `Shift + P` can show/hide its UI anywhere after registering commands.
+ */
+class Prompt {
+  private window: Window
+  private document: Document
+  private keyset: XUL.Element
+  private log: Function;
+  /**
+   * Record the last text entered
+   */
+  private lastInputText = ""
+  /**
+   * Default placeholder
+   */
+  private placeholder = "Select a command..."
+  /**
+   * Placeholder when the search result is none.
+   */
+  private emptyText = "No commands found."
+  /**
+   * It controls the max line number of commands displayed in `commandsNode`.
+   */
+  private maxLineNum: number = 12
+  /**
+  * The top-level HTML div node of `Prompt`
+  * 
+  * Changing `promptNode.style.diaplay` attribute value can show/hide `Prompt` UI.
+  */
+  public promptNode: HTMLDivElement
+  /**
+   * The HTML input node of `Prompt`.
+   * 
+   * `inputNode.placeholder` can set/get inputNode's placeholder attribute value.
+   * 
+   * `inputNode.value` can set/get inputNode's value.
+   */
+  public inputNode: HTMLInputElement
+  /**
+   * The HTML div node of `Prompt`.
+   * 
+   * `NewView` Task could create a new HTML node and then append to it.
+   */
+  public commandsNode: HTMLDivElement
+  /**
+   * Save values for SetValueTask
+   */
+  public values: string[] = null
```

Comment:
Create methods for them or get/set for them. Not all of these props need to be public.